Zend/zend_API.c

```diff
@@ -1967,24 +1967,32 @@ ZEND_API void zend_check_magic_method_implementation(const zend_class_entry *ce,
 			zend_error(error_type, "Method %s::%s() must take exactly 1 argument", ZSTR_VAL(ce->name), ZEND_GET_FUNC_NAME);
 		} else if (QUICK_ARG_SHOULD_BE_SENT_BY_REF(fptr, 1)) {
 			zend_error(error_type, "Method %s::%s() cannot take arguments by reference", ZSTR_VAL(ce->name), ZEND_GET_FUNC_NAME);
+		} else if (ZEND_TYPE_IS_SET(fptr->common.arg_info[0].type) && ZEND_TYPE_CODE(fptr->common.arg_info[0].type) != IS_STRING) {
+			zend_error(error_type, "Type of first argument of %s::%s() must be string", ZSTR_VAL(ce->name), ZEND_GET_FUNC_NAME);
```

Comment:
@nikic How do I check for types now after #4838 been implemented?

Reply:
@carusogabriel As far as I can tell you, you have to retrieve the type mask first (one version: `uint32_t type_mask = ZEND_TYPE_FULL_MASK(type);`) and use bitwise operations: `if (type_mask & MAY_BE_INT) {`. But you'll find plenty of other examples in `zend_execute` and `zend_compile` for example :)

Reply:
I think `ZEND_TYPE_CODE(x) != IS_STRING` should now be `ZEND_TYPE_PURE_MASK(x) != MAY_BE_STRING`.

Reply:
Oh right, forgot to mention that equality is what you need (instead of bitwise operations) when you check if a type exactly matches a single type, just as Christoph wrote.